whois-endtoend/src/test/java/net/ripe/db/whois/query/InverseQueryTestIntegration.java

```diff
@@ -266,7 +275,79 @@ public void inverse_email_from_untrusted_then_fail() {
         assertThat(response, containsString("is not an inverse searchable attribute"));
     }
 
+    @Test
+    public void inverse_email_UTF8_charset() {
+        databaseHelper.addObject(RpslObject.parse("" +
+                "mntner:      OWNER1-MNT\n" +
+                "descr:       Owner Maintainer\n" +
+                "admin-c:     PP1-TEST\n" +
+                "upd-to:      noreply@ripe.net\n" +
+                "auth:        MD5-PW $1$d9fKeTr2$Si7YudNf4rUGmR71n/cqk/ #test\n" +
+                "auth:        SSO person@net.net\n" +
+                "auth:        PGPKEY-A8D16B70\n" +
+                "mnt-by:      OWNER-MNT\n" +
+                "source:      TEST"));
+
+        final String rpslObject = "" +
+                "person:         Pauleth Palthen\n" +
+                "address:        Singel 258\n" +
+                "phone:          +31-1234567890\n" +
+                "remarks:        é, Ú, ß\n" +
+                "mnt-by:         OWNER1-MNT\n" +
+                "nic-hdl:        PP2-TEST\n" +
+                "remarks:        remark\n" +
+                "source:         TEST";
+
+        /*
+        * UTF-8: é(C3 89), Ú(C3 9A), ß(C3 9F)
+        * LATIN-1: é(E9), Ú(DA), ß(DF)
+        * */
+        databaseHelper.addObject(rpslObject);
+
+        final Pattern pattern = Pattern.compile("remarks:\\s+(.*?)\\s*$", Pattern.MULTILINE);
+        final String latin1ExpectedResult = getCharsetValuesInHex("é, Ú, ß", StandardCharsets.ISO_8859_1);
+        final String utf8ExpectedResult = getCharsetValuesInHex("é, Ú, ß", StandardCharsets.UTF_8);
+
+        /* Default encoding */
+        final String defaultEncodingResponse = query("-Bi mnt-by OWNER1-MNT");
+        Matcher matcher = pattern.matcher(defaultEncodingResponse);
+        if (matcher.find()){
+            assertThat(latin1ExpectedResult, is(getCharsetValuesInHex(matcher.group(1), StandardCharsets.ISO_8859_1)));
+        }
+
+        /* Latin-1 encoding */
+        final String latin1Response = query("-Z latin1 -Bi mnt-by OWNER1-MNT");
+        matcher = pattern.matcher(latin1Response);
+        if (matcher.find()){
+            assertThat(latin1ExpectedResult, is(getCharsetValuesInHex(matcher.group(1), StandardCharsets.ISO_8859_1)));
+        }
+
+        /* UTF-8 encoding */
+        final String utf8Response = query("-Z utf8 -Bi mnt-by OWNER1-MNT", StandardCharsets.UTF_8);
+        matcher = pattern.matcher(utf8Response);
+        if (matcher.find()){
+            assertThat(utf8ExpectedResult, is(getCharsetValuesInHex(matcher.group(1), StandardCharsets.UTF_8)));
+        }
+    }
+
+    private String getCharsetValuesInHex(final String values, final Charset charset){
+        byte[] utf8Bytes = String.valueOf(values).getBytes(charset);
```

Comment:
rename variable it's not necessarily utf8

Reply:
Done